src/modules/vtol_att_control/tiltrotor.cpp

```diff
@@ -387,3 +388,19 @@ void Tiltrotor::fill_actuator_outputs()
 			_actuators_fw_in->control[actuator_controls_s::INDEX_YAW];
 	}
 }
+
+/*
+ * Increase combined thrust of MC propellers if motors are tilted. Assumes that all MC motors are tilted equally.
+ */
+
+float Tiltrotor::thrust_compensation_for_tilt()
+{
+
+	// only compensate for tilt angle up to 0.5 * max tilt
+	float compensated_tilt = _tilt_control;
+	compensated_tilt = compensated_tilt < 0.0f ? 0.0f : compensated_tilt;
+	compensated_tilt = compensated_tilt > 0.5f ? 0.5f : compensated_tilt;
+
+	return _v_att_sp->thrust_body[2] / cosf(compensated_tilt * M_PI_2_F);
```

Comment:
@sfuhrer We should probably constrain this to [0,1]

Reply:
yes right, added this constrain